### PR TITLE
fix: add support for pipeline & job scope

### PR DESCRIPTION
**Pull request by d2lam**
Adding support for pipeline & job scope cache
Related: 
https://github.com/screwdriver-cd/screwdriver/issues/1284
https://github.com/screwdriver-cd/screwdriver/issues/1257

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,6 +3,26 @@
...
+ * @param  {Array}        cache   Array of cacheKeys
+ * @param  {String}       scope   Cache scope
+ * @param  {String}       action  Set or Get
+ * @return {String}       Commands to call store-cli
+ */
+function getCacheCommands(cache, scope, action) {
+    if (cache && cache.length > 0) {
+        const cmds = cache.map(item =>
+      `store-cli ${action} ${item} --type=cache --scope=${scope} || true`
+    );
```
Comment (pranavrc) [inline index.js:18]: indentation looks kinda weird here?

File: index.js [hunk 2]
```diff
@@ -37,14 +57,14 @@ class CacheBookend extends BookendInterface {
...
      */
     getTeardownCommand(o) {
-        const eventCache = hoek.reach(o.job, 'permutations.0.cache.event');
+        const cache = hoek.reach(o.job, 'permutations.0.cache');
 
-        if (eventCache && eventCache.length > 0) {
-            const eventMap = eventCache.map(item =>
-                      `store-cli set ${item} --type=cache --scope=event || true`
-                  );
+        if (cache) {
```
Comment (pranavrc) [inline index.js:62]: Would be worth adding `cache.length > 0` check here just in case, but shouldn't be a problem as is, i guess

**Review by pranavrc — APPROVED** [review]
image

File: index.js (hunk 2 above)
Comment (d2lam) [reply index.js:62]: This one is an object get from config parser. It's either there or not. The check for length is inside the getCacheCommands method.

**Pull request merged**